test/common_nn.py

```diff
@@ -742,15 +750,17 @@ def __call__(self, test_case):
         str(module)
 
         if self.reference_fn is not None:
-            out = test_case._forward_criterion(module, input, self.target)
+            out = test_case._forward_criterion(module, input, self.target, reduce=get_reduce(module))
             target = self.target
             if isinstance(target, Variable):
                 target = target.data
             expected_out = self.reference_fn(deepcopy(self._unpack_input(input)),
                                              deepcopy(target), module)
             test_case.assertEqual(out, expected_out)
 
-        test_case.check_criterion_jacobian(module, input, self.target)
+        # check_criterion_jacobian only applies when loss is a scalar.
+        if get_reduce(module):
+            test_case.check_criterion_jacobian(module, input, self.target)
```

Comment:
then use regular jacobian tests if `reduce` is `False`. Right now they are skipped entirely.

It might be better to reuse the regular `ModuleTest`  class for losses that don't reduce

Reply:
Thanks! I'll looking into reusing `ModuleTest`